packages/backup/src/Helper_Script_Manager.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * The Jetpack Backup Helper Script Manager class.
+ *
+ * @package automattic/jetpack-backup
+ */
+
+namespace Automattic\Jetpack\Backup;
+
+/**
+ * Helper_Script_Manager manages installation, deletion and cleanup of Helper Scripts
+ * to assist with backing up Jetpack Sites.
+ */
+class Helper_Script_Manager {
+
+	const RELATIVE_INSTALL_LOCATIONS = array( '', 'wp-content', 'wp-content/uploads' );
+	const TEMP_DIRECTORY             = 'jetpack-temp';
+	const HELPER_HEADER              = "<?php /* JPR Helper Script */\n";
+	const EXPIRY_TIME                = 8 * 3600; // 8 hours
+
+	const README_LINES = array(
+		'These files have been put on your server by Jetpack to assist with backups and restores of your site content. They are cleaned up automatically when we no longer need them.',
+		'If you no longer have Jetpack connected to your site, you can delete them manually.',
+		'If you have questions or need assistance, please contact Jetpack Support at https://jetpack.com/support/',
+		'If you like to build amazing things with WordPress, you should visit automattic.com/jobs and apply to join the fun – mention this file when you apply!;',
+	);
+
+	const INDEX_FILE = '<?php // Silence is golden';
+
+	/**
+	 * Installs a Helper Script, and returns its filesystem path and access url.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @param string $script_body Helper Script file contents.
+	 * @return array|WP_Error     Either an array containing the path and url of the helper script, or an error.
+	 */
+	public static function install_helper_script( $script_body ) {
+		// Check that the script body contains the correct header.
+		if ( strncmp( $script_body, self::HELPER_HEADER, strlen( self::HELPER_HEADER ) ) !== 0 ) {
+			return new \WP_Error( 'invalid_helper', 'Invalid Helper Script header' );
+		}
+
+		// Create a jetpack-temp directory for the Helper Script.
+		$relative_temp_dir = self::create_temp_directory();
+		if ( is_wp_error( $relative_temp_dir ) ) {
+			return $relative_temp_dir;
+		}
+
+		// Generate a random filename, avoid clashes.
+		$max_attempts = 5;
+		for ( $attempt = 0; $attempt < $max_attempts; $attempt++ ) {
```

Comment:
The multiple attempts are to cover potential filename clashes? When would that happen since our file is prefixed with `jp-helper`?

Reply:
It's possible for multiple Transport Servers to attempt to communicate with the same site, and as a result multiple jp-helpers can get installed. Clashes are unlikely, but this ensures they never become a problem.